presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java

```diff
@@ -219,17 +219,20 @@ public boolean usesLegacyTableLayouts()
             unenforcedConstraint = clusteringPredicatesExtractor.getUnenforcedConstraints();
         }
 
-        if (handle.getPartitions().containsAll(partitionResult.getPartitions()) && handle.getClusteringKeyPredicates().equals(clusteringKeyPredicates)) {
+        Optional<List<CassandraPartition>> currentPartitions = handle.getPartitions();
+        if (currentPartitions.isPresent() &&
+                currentPartitions.get().containsAll(partitionResult.getPartitions()) &&
```

Comment:
We used to have `containsAll`, because we misleadingly initialized `handle.getPartitions()` to empty list.
Now, this should be `! equals`.

Reply:
I didn't want to change existing behavior. Added todo

Reply:
> I didn't want to change existing behavior. Added todo

where's the TODO?